### PR TITLE
Pull Request for Issue #724

### LINKED ISSUE #724: Pistol Overwatch is not displayed under Unit Flag
Looks like this function in `UIUnitFlag` is responsible for displaying the Overwatch "Eye" icon under the unit flag, and it makes an assumption that the soldier will be using only 'Overwatch' reserve action points for Overwatch, which is not the case for at least Pistol Overwatch, and potentially for any other custom Overwatch Action Point used by a weapon equipped on the soldier.

```
//This function will be spammed, so please only send changes to flash.
simulated function RealizeOverwatch(optional XComGameState_Unit NewUnitState = none)
{
    if( NewUnitState == none )
    {
        NewUnitState = XComGameState_Unit(History.GetGameStateForObjectID(StoredObjectID));
    }

    if( NewUnitState != none )
    {

        AS_SetOverwatchIcon(NewUnitState.ReserveActionPoints.Find('Overwatch') > -1);
    }
}
```
The ideal solution would be to scan the soldier's inventory for Weapon Templates and check if the unit has any of their ` Template.OverwatchActionPoint` present on the soldier, but since the dev comment indicates that "this function will be spammed a lot", it was decided to keep things humble, and expose the list of Reserve AP we're gonna check for to a config array.

None of the weapon mods I have use any other kind of custom overwatch point, so the array will include only `Overwatch` and `PistolOverwatch`  by default. I assume there is no negative repercussion to performance as opposed to just checking specifically for these two AP in the hardcoded way.

### PR DESCRIPTION
**Pull request by Iridar**
~~Fix for issue #724 is implemented and documented, but don't merge the PR just yet, as I'm still adding docs for my previous issues.~~

Fixes #724

### CODE REVIEW

**Comment by Xymanek** [comment]
Please do not introduce new features and docs for old features in same PR.
Please make a separate PR for docs for old issues

**Comment by Iridar** [comment]
As you command. Then this PR can be merged at y'all's leisure :)

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc [hunk 1]
```diff
@@ -1684,16 +1684,42 @@ simulated function Remove()
...
-		AS_SetOverwatchIcon(NewUnitState.ReserveActionPoints.Find('Overwatch') > -1);
+		//	Start Issue #724
+		/// HL-Docs: feature:DisplayCustomOverwatchActionPointOnUnitFlag; issue:724; tags:tactical,compat
+		/// The base XCOM 2 behavior is to show an Overwatch "eye" icon under the unit flag
+		/// only if the soldier has at least one "overwatch" Reserve Action Point. 
+		/// The icon is not displayed for Pistol Overwatch. This change addresses that issue,
+		/// and also allows mods that add weapons with custom Overwatch Action Point to specify that 
+		/// action point name in XComGame.ini:
+		///
+		/// ```unrealscript
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]:
```suggestion
		/// ```ini
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc [hunk 2]
```diff
@@ -1684,16 +1684,42 @@ simulated function Remove()
...
+		/// [XComGame.CHHelpers]
+		/// +ValidReserveAPForUnitFlag = "overwatch"
+		/// +ValidReserveAPForUnitFlag = "pistoloverwatch"
+		/// ```
+		///
+		/// Compatibility: If you override `UIUnitFlag::RealizeOverwatch`, your code may undo this change.
+		foreach class'CHHelpers'.default.ValidReserveAPForUnitFlag(OverwatchActionPointName)
+		{	
+			if (NewUnitState.ReserveActionPoints.Find(OverwatchActionPointName) > -1)
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]:
```suggestion
			if (NewUnitState.ReserveActionPoints.Find(OverwatchActionPointName) > INDEX_NONE)
```

Comment (Iridar) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]: That's what the original code used, I assumed it was out of a performance concern and left it as is, but if you like.

Comment (Xymanek) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]: From `Object.uc`:
```
const INDEX_NONE = -1;
```

Both "versions" will result in same thing in the compiled code, but for reading it's better to use the "named version"

**Comment by pledbrook** [comment]
Is this approved then?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc [hunk 3]
```diff
@@ -1684,16 +1684,42 @@ simulated function Remove()
...
 	{
 		NewUnitState = XComGameState_Unit(History.GetGameStateForObjectID(StoredObjectID));
 	}
 
 	if( NewUnitState != none )
 	{
-
-		AS_SetOverwatchIcon(NewUnitState.ReserveActionPoints.Find('Overwatch') > -1);
+		//	Start Issue #724
+		/// HL-Docs: feature:DisplayCustomOverwatchActionPointOnUnitFlag; issue:724; tags:tactical,compat
```
Comment (pledbrook) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]: We will hopefully have a standard approach to documenting config options at some point, but that shouldn't hold up this PR.

Comment (pledbrook) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]: Oh, but that "compat" should definitely be a "compatibility" now.

Comment (Iridar) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIUnitFlag.uc]: Done.

**Comment by Iridar** [comment]
Approved by @Musashi1584 and @pledbrook , so I assume it's ready for merge.

**Pull request merged**